lib/reek/smells/prima_donna_method.rb

```diff
@@ -28,18 +28,25 @@ def self.contexts # :nodoc:
 
       def examine_context(ctx)
         ctx.node_instance_methods.map do |method_sexp|
-          next unless method_sexp.ends_with_bang?
+          check_for_smells(method_sexp, ctx)
+        end.compact
+      end
 
-          version_without_bang = ctx.node_instance_methods.find do |sexp_item|
-            sexp_item.name.to_s == method_sexp.name_without_bang
-          end
-          next if version_without_bang
+      private
 
-          SmellWarning.new self,
-                           context: ctx.full_name,
-                           lines: [ctx.exp.line],
-                           message: "has prima donna method `#{method_sexp.name}`"
-        end.compact
+      def check_for_smells(method_sexp, ctx)
+        return unless method_sexp.ends_with_bang?
```

Comment:
Agreed! Although it’s a private method (rightly so!), not to be callable from outside – and in the case of private methods it makes sense to make them useful / simple for the local callers. But it would indeed be best to rename this method in such a case.

But I’m ok with leaving the guard check (and the name). If there are no more nitpicks I’d love to merge this.